### DISCUSSION TITLE
[Section 3.13 - Variaveis] Dúvidas sobre variáveis de senha

### DISCUSSION THREAD

**Discussion by eduardocortez3** [chgasparoto/curso-aws-com-terraform#109] [Q&A]
Olá Cleber.

Gostaria de uma orientação quanto as variáveis que possuem senha ou dados sensíveis, qual é a forma mais segura de se trabalhar com esse tipo de informação?
Por exemplo, no meu projeto estou criando um recurso do tipo RDS, e nele preciso informar a senha e o usuário root do banco. 
1. Posso usar o AWS Secret? Como devo proceder? Tem algum exemplo?
2. Posso gerar essa senha randomicamente? Como devo proceder?
Att.

**Comment by chgasparoto** [comment]
@eduardocortez3 , o ideal é utilizar o AWS Secret ou Parameter Store ou ainda algo mais sofisticado como o Hashicorp Vault.

Eu não tenho nenhum exemplo mas se você seguir a documentação, você consegue reter esses valores utilizando este data source ([aws_ssm_parameter](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameter)) do terraform.
